### PR TITLE
JP-3859: Apply code style rules to wavecorr

### PR DESCRIPTION
**Pull request by melanieclarke**
Partially addresses [JP-3859](https://jira.stsci.edu/browse/JP-3859)

Clean up code style in wavecorr.

Also fixed a few minor logic errors in edge cases and added a few unit tests to complete coverage.

## Tasks
- [x] **request a review from someone specific**, to avoid making the maintainers review every PR
- [x] add a build milestone, i.e. `Build 11.3` (use the [latest build](https://github.com/spacetelescope/jwst/milestones) if not sure)
- [x] Does this PR change user-facing code / API? (if not, label with `no-changelog-entry-needed`)
  - [x] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types) 
  - [x] update or add relevant tests
  - [x] update relevant docstrings and / or `docs/` page
  - [x] [start a regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/jwst.yml) and include a link to the running job ([click here for instructions](https://github.com/spacetelescope/RegressionTests/blob/main/docs/running_regression_tests.md))
    - [x] Do truth files need to be updated ("okified")?
      - [ ] **after the reviewer has approved these changes**, run `okify_regtests` to update the truth files
- [ ] if a JIRA ticket exists, [make sure it is resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

<details><summary>news fragment change types...</summary>

- ``changes/<PR#>.general.rst``: infrastructure or miscellaneous change
- ``changes/<PR#>.docs.rst``
- ``changes/<PR#>.stpipe.rst``
- ``changes/<PR#>.datamodels.rst``
- ``changes/<PR#>.scripts.rst``
- ``changes/<PR#>.set_telescope_pointing.rst``
- ``changes/<PR#>.pipeline.rst``

## stage 1
- ``changes/<PR#>.group_scale.rst``
- ``changes/<PR#>.dq_init.rst``
- ``changes/<PR#>.emicorr.rst``
- ``changes/<PR#>.saturation.rst``
- ``changes/<PR#>.ipc.rst``
- ``changes/<PR#>.firstframe.rst``
- ``changes/<PR#>.lastframe.rst``
- ``changes/<PR#>.reset.rst``
- ``changes/<PR#>.superbias.rst``
- ``changes/<PR#>.refpix.rst``
- ``changes/<PR#>.linearity.rst``
- ``changes/<PR#>.rscd.rst``
- ``changes/<PR#>.persistence.rst``
- ``changes/<PR#>.dark_current.rst``
- ``changes/<PR#>.charge_migration.rst``
- ``changes/<PR#>.jump.rst``
- ``changes/<PR#>.clean_flicker_noise.rst``
- ``changes/<PR#>.ramp_fitting.rst``
- ``changes/<PR#>.gain_scale.rst``

## stage 2
- ``changes/<PR#>.assign_wcs.rst``
- ``changes/<PR#>.badpix_selfcal.rst``
- ``changes/<PR#>.msaflagopen.rst``
- ``changes/<PR#>.nsclean.rst``
- ``changes/<PR#>.imprint.rst``
- ``changes/<PR#>.background.rst``
- ``changes/<PR#>.extract_2d.rst``
- ``changes/<PR#>.master_background.rst``
- ``changes/<PR#>.wavecorr.rst``
- ``changes/<PR#>.srctype.rst``
- ``changes/<PR#>.straylight.rst``
- ``changes/<PR#>.wfss_contam.rst``
- ``changes/<PR#>.flatfield.rst``
- ``changes/<PR#>.fringe.rst``
- ``changes/<PR#>.pathloss.rst``
- ``changes/<PR#>.barshadow.rst``
- ``changes/<PR#>.photom.rst``
- ``changes/<PR#>.pixel_replace.rst``
- ``changes/<PR#>.resample_spec.rst``
- ``changes/<PR#>.residual_fringe.rst``
- ``changes/<PR#>.cube_build.rst``
- ``changes/<PR#>.extract_1d.rst``
- ``changes/<PR#>.resample.rst``

## stage 3
- ``changes/<PR#>.assign_mtwcs.rst``
- ``changes/<PR#>.mrs_imatch.rst``
- ``changes/<PR#>.tweakreg.rst``
- ``changes/<PR#>.skymatch.rst``
- ``changes/<PR#>.exp_to_source.rst``
- ``changes/<PR#>.outlier_detection.rst``
- ``changes/<PR#>.tso_photometry.rst``
- ``changes/<PR#>.stack_refs.rst``
- ``changes/<PR#>.align_refs.rst``
- ``changes/<PR#>.klip.rst``
- ``changes/<PR#>.spectral_leak.rst``
- ``changes/<PR#>.source_catalog.rst``
- ``changes/<PR#>.combine_1d.rst``
- ``changes/<PR#>.ami.rst``

## other
- ``changes/<PR#>.wfs_combine.rst``
- ``changes/<PR#>.white_light.rst``
- ``changes/<PR#>.cube_skymatch.rst``
- ``changes/<PR#>.engdb_tools.rst``
- ``changes/<PR#>.guider_cds.rst``
</details>

### CODE REVIEW

**Comment by melanieclarke** [comment]
Regression tests here:
https://github.com/spacetelescope/RegressionTests/actions/runs/13082159082

File: jwst/wavecorr/wavecorr_step.py [hunk 1]
```diff
@@ -65,11 +96,7 @@ def process(self, step_input):
...
 
 def _check_slit_metadata_attributes(slit):
-
-    if not hasattr(slit.meta, 'wcs') and slit.meta.wcs is not None:
-        raise AttributeError("Input model does not have a WCS object; assign_wcs should "
-                             "be run before wavecorr.")
-
-    if not hasattr(slit, 'source_xpos'):
-        raise AttributeError("Input model does not have source_xpos attribute; "
-                             "extract_2d should be run before wavecorr.")
```
Comment (emolter) [inline jwst/wavecorr/wavecorr_step.py:75]: What made this check no longer necessary?

File: jwst/wavecorr/wavecorr_step.py [hunk 2]
```diff
@@ -9,36 +8,66 @@
 
 class WavecorrStep(Step):
     """
-    This step applies wavelength offsets to off-center NIRSpec sources.
+    Apply wavelength offsets to off-center NIRSpec sources.
+
+    Wavelength corrections are applied only to point sources in NIRSpec MOS
+    and FS data.
+
+    The algorithm uses a reference file which is a look-up table of
```
Comment (emolter) [inline jwst/wavecorr/wavecorr_step.py:16]:
```suggestion
    The algorithm uses a reference file that is a look-up table of
```

**Review by emolter — APPROVED** [review]
LGTM, just one question and a tiny grammar thing.

Thanks for adding some unit test coverage, too!

File: jwst/wavecorr/wavecorr_step.py (hunk 1 above)
Comment (melanieclarke) [reply jwst/wavecorr/wavecorr_step.py:75]: This is unreachable for acceptable datamodels - the slit will always have source_xpos, whether or not it was populated by extract_2d.  Since the default value is 0.0, and 0.0 is also a legitimate value for the source x-position, there's no way to know from the value if source_xpos was correctly set or not.  The extract_2d step is required to get the data into MultiSlit format anyway, so it would be pretty difficult to get here without running it.

**Comment by melanieclarke** [comment]
Running regression tests again to make sure nothing new has come up:
https://github.com/spacetelescope/RegressionTests/actions/runs/13288253594

**Review by tapastro — APPROVED** [review]
LGTM
